after_sign_hook.js

```diff
@@ -1,46 +1,45 @@
-const fs = require('fs');
-const path = require('path');
-var electron_notarize = require('electron-notarize');
-
-module.exports = async function(params) {
-  // Only notarize the app on Mac OS only.
-  if (params.electronPlatformName !== 'darwin') {
-    return;
-  }
-  console.log('afterSign hook triggered', params);
-
-  if (!process.env.CIRCLE_TAG || process.env.CIRCLE_TAG.length === 0) {
-    console.log('Not on a tag. Skipping notarization');
-    return;
-  }
-
-  // Same appId in electron-builder.
-  let appId = 'com.automattic.wordpress';
-
-  let appPath = params.appOutDir
-    ? path.join(
-        params.appOutDir,
-        `${params.packager.appInfo.productFilename}.app`
-      )
-    : params.artifactPaths[0].replace(new RegExp('.blockmap'), '');
-
-  if (!fs.existsSync(appPath)) {
-    throw new Error(`Cannot find application at: ${appPath}`);
-  }
-
-  console.log(`Notarizing ${appId} found at ${appPath}`);
-
-  try {
-    await electron_notarize.notarize({
-      appBundleId: appId,
-      appPath: appPath,
-      appleId: process.env.NOTARIZATION_ID,
-      appleIdPassword: process.env.NOTARIZATION_PWD,
-      ascProvider: 'AutomatticInc',
-    });
-  } catch (error) {
-    console.error(error);
-  }
-
-  console.log(`Done notarizing ${appId}`);
-};
+#!/usr/bin/env node
+
+const path = require( 'path' );
+const { notarize } = require( 'electron-notarize' );
+
+const APP_ID = 'com.automattic.wordpress';
+const NOTARIZATION_ASC_PROVIDER = 'AutomatticInc';
+const NOTARIZATION_ID = process.env.NOTARIZATION_ID;
+const NOTARIZATION_PWD = process.env.NOTARIZATION_PWD;
+
+const shouldNotarize = process.platform === 'darwin' && !!process.env.CIRCLE_TAG;
+
+function elapsed( start ) {
+	const now = new Date();
+
+	const ms = Math.abs( now.getTime() - start.getTime() );
+	const diff = new Date( ms );
+
+	return `${ diff.getMinutes() } minutes, ${ diff.getSeconds() } seconds`;
+}
+
+module.exports = async function( context ) {
+	if ( ! shouldNotarize ) {
+		return;
+	}
+
+	try {
+		const app = path.join( context.appOutDir, `${ context.packager.appInfo.productFilename }.app` );
+		const appName = path.basename( app );
```

Comment:
we should probably move these outside of the `try` since we don't expect them to fail, yeah?

Reply:
Yep - makes sense. In the unlikely event these constants are undefined the logic an error will be thrown any way. Updated in bb56851